### PR TITLE
Migrate google_compute_firewall_policy_rule resource from DCL to MMv1

### PR DESCRIPTION
**Pull request by Samir-Cit**
Hello folks.
This PR is to move the Firewall Policy Rule (`google_compute_firewall_policy_rule`) resource from DCL to MMv1. And this PR includes:
- Creation of the resource on MMv1.
- Creation of the examples and tests.
- Remove references to DCL resource.

_**Note:** This PR doesn't include the deletion of the DCL resource. This change will be done on the [Delarative Resource Client Library](https://github.com/GoogleCloudPlatform/declarative-resource-client-library/tree/main) repository._

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: `google_compute_firewall_policy_rule` now uses MMv1 engine instead of DCL.
```

### CODE REVIEW

File: mmv1/products/compute/FirewallPolicyRule.yaml [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'FirewallPolicyRule'
+kind: 'compute#firewallPolicyRule'
+base_url: 'locations/global/firewallPolicies/{{firewall_policy}}'
```
Comment (BBBmau) [inline mmv1/products/compute/FirewallPolicyRule.yaml:17]: The error comes from having the url being nested in itself.

From the logs we can see:
```hcl
---[ REQUEST ]---------------------------------------
POST /compute/beta/locations/global/firewallPolicies/locations/global/firewallPolicies/79416795910/addRule?alt=json HTTP/1.1
Host: compute.googleapis.com
User-Agent: Terraform/1.8.3 (+https://www.terraform.io) Terraform-Plugin-SDK/2.33.0 terraform-provider-google-beta/acc
Content-Length: 458
Content-Type: application/json
Accept-Encoding: gzip

{
 "action": "deny",
 "description": "Resource created for Terraform acceptance testing",
 "direction": "INGRESS",
 "match": {
  "layer4Configs": [
   {
    "ipProtocol": "tcp",
    "ports": [
     "80",
     "8080"
    ]
   },
   {
    "ipProtocol": "all"
   }
  ],
  "srcAddressGroups": [
   "organizations/529579013760/locations/global/addressGroups/tf-test-policykir3yqimoi"
  ],
  "srcFqdns": [
   "google.com"
  ],
  "srcIpRanges": [
   "11.100.0.1/32"
  ],
  "srcRegionCodes": [
   "US"
  ],
  "srcThreatIntelligences": [
   "iplist-known-malicious-ips"
  ]
 },
 "priority": 9001
}

-----------------------------------------------------
2024/08/09 23:44:32 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 404 Not Found
Content-Length: 1670
Content-Type: text/html; charset=UTF-8
Date: Fri, 09 Aug 2024 23:44:32 GMT
Server: ESF
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0
```

What to look for is the POST url:`POST /compute/beta/locations/global/firewallPolicies/locations/global/firewallPolicies/79416795910/addRule`

Where `{{firewall_policy}} = locations/global/firewallPolicies/79416795910` in `locations/global/firewallPolicies/{{firewall_policy}}`

tl;dr you'll want to just use `{{firewall_policy}}` for the urls here instead of what's set currently. Let me know if something doesn't make sense from this explanation.

Comment (Samir-Cit) [reply mmv1/products/compute/FirewallPolicyRule.yaml:17]: Oh, I did the fix for this error but I forgot to push the changes.
I'll push in a couple of minutues!
Thanks

**Comment by Samir-Cit** [comment]
> /gcbrun

@BBBmau  this command is working?
Looks like it didn't execute any check.

File: mmv1/templates/terraform/examples/firewall_policy_rule.tf.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 resource "google_network_security_address_group" "basic_global_networksecurity_address_group" {
```
Comment (BBBmau) [inline mmv1/templates/terraform/examples/firewall_policy_rule.tf.erb:1]: could you address the merge conflicts that are found in this file?

Comment (Samir-Cit) [reply mmv1/templates/terraform/examples/firewall_policy_rule.tf.erb:1]: Sure, just did!

**Review by trodge — CHANGES_REQUESTED** [review]

**Comment by c2thorn** [comment]
Confirming that I have spoken with @Samir-Cit offline and received permission to force push to this branch

**Comment by trodge** [comment]
Backup located [here](https://github.com/trodge/magic-modules/tree/feat/FirewallPolicyRule-11360-backup-0)

**Review by trodge — APPROVED** [review]
